### PR TITLE
Styled components + front migration [CU-6mtkw8]

### PR DESCRIPTION
**Pull request by Fuzznimp**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,23 +16,43 @@
     "graphql-tag": "2.10.4",
     "next": "9.4.4",
     "react": "16.13.1",
-    "react-dom": "16.13.1"
+    "react-dom": "16.13.1",
+    "react-feather": "2.0.8",
```
Comment (Fenntasy) [inline package.json:20]: Just putting a comment that came to mind but it's not blocking for this PR.
I'm not against `react-feather` in principle but in the future, it would be great to get rid of it.

Here is my thinking: we need to manage a repo that will be forked by other people that will add their design systems on top of it. It would then be great to not "impose" a dependency that these other system would use.

But then again, this PR is already big so let's not do it now ^^

File: src/command/addIdentityToUser.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    $type: IdentityTypes!
+    $value: String!
+  ) {
+    addIdentityToUser(input: { userId: $userId, type: $type, value: $value }) {
+      id
+      primary
+      status
+      type
+      value
+      __typename
```
Comment (Fenntasy) [inline src/command/addIdentityToUser.ts:19]: curiosity, why is that needed?

File: src/components/InputForm.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import React from "react";
+import styled from "styled-components";
+
+import { IdentityTypes } from "../@types/Identity";
+
+export const InputForm: React.FC<{ type: IdentityTypes }> = ({ type }) => {
+  const [formValue, setFormValue] = React.useState("");
```
Comment (Fenntasy) [inline src/components/InputForm.tsx:7]: This is more of a `EmailIdentityForm` with `const [email, setEmail] = React.useState("")` don't you think?

File: src/design-system/theme/useTheme.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { useContext } from "react";
+import { ThemeContext } from "styled-components";
+
+import { LightTheme } from "./lightTheme";
```
Comment (Fenntasy) [inline src/design-system/theme/useTheme.ts:4]: Thinking out loud, this would be a good use case for having a default export in a `theme.ts` file.
We could check in it a Config and return either the light theme as a default or the theme set in the config.
Having a default export would allow us to not name it since we wouldn't know which one is selected

File: src/pages/api/identity.tsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      })
+      .catch((error) => {
+        console.log(error);
+
+        response.statusCode = 400;
+
+        return response.json("failed");
+      });
+  }
+  response.statusCode = 400;
```
Comment (Fenntasy) [inline src/pages/api/identity.tsx:41]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405

And can we do something like `response.status(405).end()`?

File: src/pages/api/identity.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+  return response.json(":(");
```
Comment (Fenntasy) [inline src/pages/api/identity.tsx:43]:
```suggestion
  return response.end();
```

File: src/pages/api/identity.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        response.statusCode = 200;
+
+        return response.json("succeed");
+      })
+      .catch((error) => {
+        console.log(error);
+
+        response.statusCode = 400;
+
+        return response.json("failed");
```
Comment (Fenntasy) [inline src/pages/api/identity.tsx:38]:
```suggestion
        return response.json({error: "Could not remove identity"});
```

File: src/pages/api/identity.tsx [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
```
Comment (Fenntasy) [inline src/pages/api/identity.tsx:24]:
```suggestion
        return response.json({error: "Could not add identity"});
```

File: src/pages/api/identity.tsx [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+  response: NextApiResponse,
+): Promise<void> => {
+  const { userId, type, value } = request.body;
+
+  if (request.method === "POST") {
+    return await addIdentityToUser({ userId, type, value })
+      .then(() => {
+        response.statusCode = 200;
+
+        return response.json("succeed");
```
Comment (Fenntasy) [inline src/pages/api/identity.tsx:17]: The custom is usually to return either the added thing (and that could help you don't need to refresh the page)

File: src/components/InputForm.tsx (hunk 3 above)
Comment (Fuzznimp) [reply src/components/InputForm.tsx:7]: Yeah true, I wanted to have a global component but it's not needed there. I'll change it to make it more readable.

File: src/command/addIdentityToUser.ts (hunk 2 above)
Comment (Fuzznimp) [reply src/command/addIdentityToUser.ts:19]: For now, none of these are need (even the ID, as we don't use apollo caching). Do you want me to remove them?

File: src/pages/api/identity.tsx (hunk 5 above)
Comment (Fuzznimp) [reply src/pages/api/identity.tsx:41]: Good error code to know. I can't use status as a method though, it's typed as number.

```ts
    class ServerResponse extends OutgoingMessage {
        statusCode: number;
        statusMessage: string;

        constructor(req: IncomingMessage);

        assignSocket(socket: Socket): void;
        detachSocket(socket: Socket): void;
        // https://github.com/nodejs/node/blob/master/test/parallel/test-http-write-callbacks.js#L53
        // no args in writeContinue callback
        writeContinue(callback?: () => void): void;
        writeHead(statusCode: number, reasonPhrase?: string, headers?: OutgoingHttpHeaders): this;
        writeHead(statusCode: number, headers?: OutgoingHttpHeaders): this;
        writeProcessing(): void;
    }
```

Comment (Fuzznimp) [reply src/pages/api/identity.tsx:41]: So I'll just change the error code (the error message has been changed in the next pr, so I won't change it here)

File: src/pages/api/identity.tsx (hunk 7 above)
Comment (Fuzznimp) [reply src/pages/api/identity.tsx:38]: Ok nvm, I'll change them there.

File: src/pages/api/identity.tsx (hunk 9 above)
Comment (Fuzznimp) [reply src/pages/api/identity.tsx:17]: Quick question, as the `response` object is massive. Do I have the data added in the `response` object ? Or do I, if succeed, return what was in the payload?

Comment (Fuzznimp) [reply src/pages/api/identity.tsx:17]: I.e.
```ts
return await addIdentityToUser({ userId, type, value })
      .then((data) => {
        response.statusCode = 200;

        return response.json({data});
      })
```

Comment (Fenntasy) [reply src/pages/api/identity.tsx:17]: I think you may need some adjustments on that, depending on what is the return. 

Take care also that errors in a graphQL request are sent as a success in HTTP so I think we may need to check if there are errors in there

Comment (Fuzznimp) [reply src/pages/api/identity.tsx:17]: > Take care also that errors in a graphQL request are sent as a success in HTTP so I think we may need to check if there are errors in there 

Yeah I realised that while building the app. Do you want me to make the change here? Maybe a new task would be wiser.

Comment (Fenntasy) [reply src/pages/api/identity.tsx:17]: Go for a new task

File: src/design-system/theme/useTheme.ts (hunk 4 above)
Comment (Fuzznimp) [reply src/design-system/theme/useTheme.ts:4]: I see the idea, but I think we should talk about it to be sure I get the full picture.

Comment (Fuzznimp) [reply src/design-system/theme/useTheme.ts:4]: Made the task

File: src/components/EmailIdentityForm.tsx [hunk 10]
```diff
@@ -3,7 +3,9 @@ import styled from "styled-components";
 
 import { IdentityTypes } from "../@types/Identity";
 
-export const InputForm: React.FC<{ type: IdentityTypes }> = ({ type }) => {
+export const EmailIdentityForm: React.FC<{ type: IdentityTypes }> = ({
+  type,
+}) => {
   const [formValue, setFormValue] = React.useState("");
```
Comment (Fenntasy) [inline src/components/EmailIdentityForm.tsx:9]: Well, if you made the change here, what about 

`const [email, setEmail] = React.useState("")` ? 😇

Comment (Fuzznimp) [reply src/components/EmailIdentityForm.tsx:9]: Woops 🙇

Comment (Fuzznimp) [reply src/components/EmailIdentityForm.tsx:9]: Wait, actually, this is also used to add/remove phone number 🤦‍♂️. Sorry, I'm not awake.

```ts
<Flex>
            <Button
              onClick={() => setAddPhone(!addEmail)}
              color={theme.colors.green}
            >
              Add a phone number
            </Button>
            {addPhone && <EmailIdentityForm type={IdentityTypes.PHONE} />}
          </Flex>
```

Comment (Fuzznimp) [reply src/components/EmailIdentityForm.tsx]: f6daa83

File: src/components/IdentityInputForm.tsx [hunk 11]
```diff
@@ -3,19 +3,19 @@ import styled from "styled-components";
 
 import { IdentityTypes } from "../@types/Identity";
 
-export const EmailIdentityForm: React.FC<{ type: IdentityTypes }> = ({
+export const IdentityInputForm: React.FC<{ type: IdentityTypes }> = ({
```
Comment (Fenntasy) [inline src/components/IdentityInputForm.tsx:6]: 👍
